src/client.ts

```diff
@@ -335,6 +341,152 @@ export class McpdClient {
+   *
+   * Servers that don't implement prompts (return 501 Not Implemented) are silently
+   * skipped, allowing this method to work with mixed server types.
+   *
+   * Prompt fetches from multiple servers are executed concurrently for optimal performance.
+   *
+   * @param options - Optional configuration
+   * @param options.servers - Array of server names to include. If not specified, includes all servers.
+   * @returns Array of prompt schemas with transformed names (serverName__promptName). Only includes prompts from healthy servers that support them.
+   * @throws {ConnectionError} If unable to connect to the mcpd daemon
+   * @throws {TimeoutError} If requests to the daemon time out
+   * @throws {AuthenticationError} If API key authentication fails
+   * @throws {McpdError} If health check or initial server listing fails
+   *
+   * @example
+   * ```typescript
+   * // Get all prompts from all servers
+   * const allPrompts = await client.getPrompts();
+   * // Returns: [
+   * //   { name: "github__create_pr", description: "...", arguments: [...] },
+   * //   { name: "notion__create_page", description: "...", arguments: [...] }
+   * // ]
+   *
+   * // Get prompts from specific servers only
+   * const somePrompts = await client.getPrompts({ servers: ['github', 'notion'] });
+   *
+   * // Original prompt name "create_pr" becomes "github__create_pr"
+   * // This prevents clashes if multiple servers have prompts with the same name
+   * ```
+   */
+  async getPrompts(options?: { servers?: string[] }): Promise<Prompt[]> {
+    const { servers } = options || {};
+
+    // Determine which servers to query.
+    const serverNames =
+      servers && servers.length > 0 ? servers : await this.listServers();
+
+    // Get health status for all servers.
+    const healthMap = await this.getServerHealth();
+
+    // Filter to only healthy servers.
+    const healthyServers = serverNames.filter((name) => {
+      const health = healthMap[name];
+      return health && HealthStatusHelpers.isHealthy(health.status);
+    });
+
+    // Fetch prompts from all healthy servers in parallel.
+    const results = await Promise.allSettled(
+      healthyServers.map(async (serverName) => ({
+        serverName,
+        prompts: await this.#getPromptsByServer(serverName),
+      })),
+    );
+
+    // Process results and transform prompt names.
+    const allPrompts: Prompt[] = [];
+    for (const result of results) {
+      if (result.status === "fulfilled") {
+        const { serverName, prompts } = result.value;
+        // Transform prompt names to serverName__promptName format.
+        for (const prompt of prompts) {
+          allPrompts.push({
+            ...prompt,
+            name: `${serverName}__${prompt.name}`,
+          });
+        }
+      } else {
+        // If we can't get prompts for a server, skip it with a warning.
+        console.warn(`Failed to get prompts for server:`, result.reason);
+      }
+    }
```

Comment:
nit: to avoid multiple nesting levels and generic `for`, `forEach` loops, using `.map` `.filter` can make the code easier to read and side effects free


```suggestion
const allPrompts: Prompt[] = results.flatMap(result => {
  if (result.status === "fulfilled") {
    const { serverName, prompts } = result.value;
    return prompts.map(prompt => ({
      ...prompt,
      name: `${serverName}__${prompt.name}`,
    }));
  } else {
    console.warn(`Failed to get prompts for server:`, result.reason);
    return []; // Return an empty array for rejected promises
  }
});